screen-orientation/lock-basic.html

```diff
@@ -27,17 +26,27 @@
     'landscape-secondary',
     'portrait-primary',
   ];
-  const promises = orientations.map(type => {
-    const returnType = screen.orientation.lock(type);
+  for (const orientation of orientations) {
+    const returnType = screen.orientation.lock(orientation);
     assert_true(returnType instanceof Promise);
-  });
-  return Promise.all(promises).finally(() => {
-    screen.orientation.unlock();
-  });
+    await returnType;
+  }
+  screen.orientation.unlock();
 }, "Test that screen.orientation.lock returns a pending promise.");
 
-test(() => {
-  const previousOrientation = screen.orientation;
-  assert_equals(screen.orientation, previousOrientation);
+promise_test(async t => {
+  const preType = screen.orientation.type;
+  let newType = "";
```

Comment:
Nit - the `preType.indexOf("portrait") != -1)` is confusing, maybe: 
 
```JS
const isPortrait = screen.orientation.type.includes("portrait");
const newType = `${ isPortrait ? "portrait" : "landscape" }-primary`;
```

Reply:
Updated.